packages/typir/src/kinds/function-kind.ts

```diff
@@ -413,9 +418,9 @@ export class FunctionKind implements Kind {
         }
         if (overloaded.overloadedFunctions.length <= 0) {
             // remember the output type of the first function
-            overloaded.sameOutputType = outputType;
+            overloaded.sameOutputType = outputTypeForFunctionCalls;
         } else {
-            if (overloaded.sameOutputType && outputType && this.typir.equality.areTypesEqual(overloaded.sameOutputType, outputType) === true) {
+            if (overloaded.sameOutputType && outputTypeForFunctionCalls && this.typir.equality.areTypesEqual(overloaded.sameOutputType, outputTypeForFunctionCalls) === true) {
                 // the output types of all overloaded functions are the same for now
```

Comment:
Yes: A few lines below, `overloaded.sameOutputType` might be reset to `undefined`. The objective of these complicated checks are to improve performance for the special case, that all overloaded functions have the same return type. In that case, no type inference for the input parameters is required, since the return/output type is always the same.

Reply:
OK, thanks